start_dev.sh

```diff
@@ -1,2 +0,0 @@
-#!/bin/sh
```

Comment:
Please maintain a shell file for jumping into the work environment that does not require `make` or any build tooling to be installed on the users computer.

Reply:
I just thought that since a user/developer is going to get into the project, in one way or another `make` will be needed at least to compile the firmware itself in the first place. Plus, on OSX `make` comes out of the box (at least it's used to be) and on GNU/Linux it's usually getting installed right with `gcc`/build-essentials packages.

Anyway, as always, I'm not arguing, just discussing thinking out loud. So I guess you talk about cases when a user have an environment not of a regular OS but more something like "container hypervisor", where everything that is allowed is just to start/stop containers with everything needed inside only. If it's the case then I agree. But is it ok if a script will be back but in `script`/ directory?

Reply:
Honestly, its mostly windows users who setup WSL+Docker and dont have any other tooling installed. The _nice_ part of Docker being used for the builds is that it keeps the host side dependencies nice and simple (git, docker, a shell). 

Script can be in the `script` folder, though at this point may argue that the `Makefile` should be there too

Reply:
> its mostly windows users who setup WSL+Docker and dont have any other tooling installed.

Aha, got it.

> The _nice_ part of Docker being used for the builds is that it keeps the host side dependencies nice and simple (git, docker, a shell).

Yeah, I heard that it's one of the reasons of its popularity nowadays.

> Script can be in the `script` folder

Deal!

> though at this point may argue that the `Makefile` should be there too

Well... the way how I see it is that the whole point of placing `Makefile` in the root of the project is to make "lazy" & "easy" "entry point" of the project for old-school tinkerers who got used to have a "main" top-level `Makefile` to manage a project. 👉 👈

Reply:
>Yeah, I heard that it's one of the reasons of its popularity nowadays.

Yeah its definitely one; but also the main reason I love it is that it forces everyone to be working in lockstep on the base base libraries and compilers. Given that there have been quite a few bugs that turned out to be the user using an older compiler or quirks with their setup.

>Well... the way how I see it is that the whole point of placing Makefile in the root of the project is to make "lazy" & "easy" "entry point" of the project for old-school tinkerers who got used to have a "main" top-level Makefile to manage a project. point_right point_left

I love this because the `start_dev.sh` in the root was basically one of the modern versions of that 🤣. Like its _for the same reason_ of being an easy start point. As a lot of people are more comfortable looking for a `build.sh` or `start.sh` than figuring out the invocation to `make`. (one these is easier to tab complete than the other).

Reply:
> there have been quite a few bugs that turned out to be the user using an older compiler

Been there, done that! For example, `arm-none-eabi-gcc` from `Ubuntu 18.04 LTS` for _some reason_ has relatively low compression ratio when compiling the firmware for Miniware models.

And I guess `alpine` has been selected since it's probably the only one modern distro which have stable binary pre-compiled GNU compiler toolchain for `RiscV`.

> figuring out the invocation to `make`

Your very valid point made me update `Makefile` in the following way:
  - now there are `help` and `list` targets
  - `help` target invoked by default 👍